### PR TITLE
feat: add source maps generation

### PR DESCRIPTION
**Pull request by guilhermedeandrade**
This PR will add the [source-map-cli](https://github.com/gabmontes/source-map-cli) dev dependency, add scripts to resolve source maps using it and also create a script to build an Android `.apk` file

### CODE REVIEW

**Comment by guilhermedeandrade** [comment]
> We need to update the "Available scripts" page of the documentation.

Done 🎉

File: docusaurus/docs/what-is-included/available-scripts.md [hunk 1]
```diff
@@ -68,6 +68,18 @@ This script will list all the available [Android Virtual Devices (AVD)](https://
 
 This script will list all the connected Android devices, either virtual (simulator) or physical.
 
+## `npm run sourcemap:ios:resolve`
+
+This script will resolve the `ios.bundle.map` file according to the given line and column.
```
Comment (acostalima) [inline docusaurus/docs/what-is-included/available-scripts.md:73]:
```suggestion
Given the stack trace created at runtime when an error occurs in the JavaScript production bundle, this script resolves the location (file, line and column) from which the error originated in the original source code. This is done by taking the line and column reported and match it is against `ios.bundle.map` source map. Useful for debugging purposes.
```

File: docusaurus/docs/what-is-included/available-scripts.md [hunk 2]
```diff
@@ -68,6 +68,18 @@ This script will list all the available [Android Virtual Devices (AVD)](https://
...
+
+Usage: `npm run sourcemap:ios:resolve 42 70`
+
+## `npm run sourcemap:android:resolve`
+
+This script will resolve the `android.bundle.map` file according to the given line and column.
```
Comment (acostalima) [inline docusaurus/docs/what-is-included/available-scripts.md:79]:
```suggestion
Given the stack trace created at runtime when an error occurs in the JavaScript production bundle, this script resolves the location (file, line and column) from which the error originated in the original source code. This is done by taking the line and column reported and match it is against `android.bundle.map` source map. Useful for debugging purposes.
```

File: docusaurus/docs/what-is-included/available-scripts.md [hunk 3]
```diff
@@ -92,6 +104,10 @@ This script will automatically open the documentation in your default browser.
 Keep in mind, you must have your documentation's dependencies installed before running this script!
 To to so, please run `npm install --prefix docusaurus` first.
 
+## `npm run prerelease`
+
+This script will run both `build:android:js` and `build:ios:js:prod` scripts, preparing for a release.
+
```
Comment (acostalima) [inline docusaurus/docs/what-is-included/available-scripts.md:110]: This script is not supposed to be ran by the developer directly. As such, we can omit it.

File: docusaurus/docs/what-is-included/available-scripts.md [hunk 4]
```diff
@@ -24,37 +24,37 @@ This script will launch the app on the first available simulator it is found. Yo
...
-You can find the bundle file, `index.android.bundle`, in the project's root directory.
+You can find the source map file, `ios.bundle.map`, in the project's root directory.
 
 ## `npm run build:android:js:prod`
 
-This script will create an optimized, production-ready JavaScript bundle targeted to the Android platform.
+This script will create an optimized, production-ready JavaScript bundle targeted to the Android platform. It will also generate a source map.
 
-You can find the bundle file, `index.android.bundle`, in the project's root directory.
+You can find the both the bundle, `index.android.bundle`, in the project's `/android/app/src/main/assets/` directory. The source map file, `android.bundle.map`, will be in the project's root directory.
```
Comment (acostalima) [inline docusaurus/docs/what-is-included/available-scripts.md:45]:
```suggestion
You will find the the bundle file, `index.android.bundle`, in the directory `android/app/src/main/assets/`. The source map file, `android.bundle.map`, is created at the project's root directory.
```

File: docusaurus/docs/what-is-included/available-scripts.md [hunk 5]
```diff
@@ -24,37 +24,37 @@ This script will launch the app on the first available simulator it is found. Yo
...
-This script will create a non-optimized JavaScript bundle targeted for debugging purposes targeted to the Android platform.
+This script will generate a source map for the iOS platform.
 
-You can find the bundle file, `index.android.bundle`, in the project's root directory.
+You can find the source map file, `ios.bundle.map`, in the project's root directory.
 
 ## `npm run build:android:js:prod`
 
-This script will create an optimized, production-ready JavaScript bundle targeted to the Android platform.
+This script will create an optimized, production-ready JavaScript bundle targeted to the Android platform. It will also generate a source map.
```
Comment (acostalima) [inline docusaurus/docs/what-is-included/available-scripts.md:43]:
```suggestion
This script will create an optimized, production-ready JavaScript bundle targeted to the Android platform and it's respective source map.
```

File: docusaurus/docs/what-is-included/available-scripts.md [hunk 6]
```diff
@@ -24,37 +24,37 @@ This script will launch the app on the first available simulator it is found. Yo
...
-## `npm run build:android:js:dev`
+## `npm run build:ios:js:prod`
 
-This script will create a non-optimized JavaScript bundle targeted for debugging purposes targeted to the Android platform.
+This script will generate a source map for the iOS platform.
 
-You can find the bundle file, `index.android.bundle`, in the project's root directory.
+You can find the source map file, `ios.bundle.map`, in the project's root directory.
 
 ## `npm run build:android:js:prod`
```
Comment (acostalima) [inline docusaurus/docs/what-is-included/available-scripts.md:41]:
```suggestion
## `npm run bundle:android:js:prod`
```

File: docusaurus/docs/what-is-included/available-scripts.md [hunk 7]
```diff
@@ -24,37 +24,37 @@ This script will launch the app on the first available simulator it is found. Yo
...
 
 ## `npm run start:android:prod`
 
 Runs the Android app in release mode.
 
-This script will launch the app on the first available connected device it is found. You can override this behavior by passing the argument `--deviceId <device id>` where `device id` can be obtained from the output of `list:android:devices` script. 
+This script will launch the app on the first available connected device it is found. You can override this behavior by passing the argument `--deviceId <device id>` where `device id` can be obtained from the output of `list:android:devices` script.
 
-## `npm run build:android:js:dev`
+## `npm run build:ios:js:prod`
```
Comment (acostalima) [inline docusaurus/docs/what-is-included/available-scripts.md:35]:
```suggestion
## `npm run bundle:ios:js:prod`
```

File: docusaurus/docs/what-is-included/available-scripts.md [hunk 8]
```diff
@@ -24,37 +24,37 @@ This script will launch the app on the first available simulator it is found. Yo
...
 
-This script will create a non-optimized JavaScript bundle targeted for debugging purposes targeted to the Android platform.
+This script will generate a source map for the iOS platform.
```
Comment (acostalima) [inline docusaurus/docs/what-is-included/available-scripts.md:37]:
```suggestion
This script will generate a JavaScript source map for the iOS platform.
```

File: docusaurus/docs/what-is-included/available-scripts.md [hunk 9]
```diff
@@ -24,37 +24,37 @@ This script will launch the app on the first available simulator it is found. Yo
...
 You can find the AAB file at `android/app/build/outputs/bundle/<release|debug>/app.aab`.
 
-## `npm run build:android:aab:staging`
+## `npm run build:android:apk`
 
-This script will create a production-ready [Android App Bundle (AAB)](https://developer.android.com/platform/technology/app-bundle) configured for the staging environment. This is the binary that you should upload to the Google Play Console to release the app.
+This script will create a production-ready Android package (APK) configured for the production environment.
 
-You can find the AAB file at `android/app/build/outputs/bundle/<release|debug>/app.aab`.
+You can find the APK file at `android/app/build/outputs/bundle/<release|debug>/app.apk`.
```
[the remark this answers is not in the document]
Comment (guilhermedeandrade) [reply docusaurus/docs/what-is-included/available-scripts.md:57]: Yes I checked

File: docusaurus/docs/what-is-included/available-scripts.md (hunk 3 above)
Comment (guilhermedeandrade) [reply docusaurus/docs/what-is-included/available-scripts.md:110]: Ok, I will remove it 👍

File: docusaurus/docs/what-is-included/available-scripts.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (acostalima) [reply docusaurus/docs/what-is-included/available-scripts.md:57]: Hmm, I was under the impression APKs were saved to `android/app/build/outputs/apk/release/app-release.apk` or something like this 🤔

[the remark this answers is not in the document]
Comment (acostalima) [reply docusaurus/docs/what-is-included/available-scripts.md:57]: I've just double checked and the path above is the correct one. 😉

File: docusaurus/docs/what-is-included/available-scripts.md [hunk 10]
```diff
@@ -24,37 +24,37 @@ This script will launch the app on the first available simulator it is found. Yo
...
 Runs the Android app in release mode.
 
-This script will launch the app on the first available connected device it is found. You can override this behavior by passing the argument `--deviceId <device id>` where `device id` can be obtained from the output of `list:android:devices` script. 
+This script will launch the app on the first available connected device it is found. You can override this behavior by passing the argument `--deviceId <device id>` where `device id` can be obtained from the output of `list:android:devices` script.
 
-## `npm run build:android:js:dev`
+## `npm run bundle:ios:js:prod`
 
-This script will create a non-optimized JavaScript bundle targeted for debugging purposes targeted to the Android platform.
+This script will generate a JavaScript source map for the iOS platform.
```
Comment (acostalima) [inline docusaurus/docs/what-is-included/available-scripts.md:37]:
```suggestion
This script will create an optimized, production-ready JavaScript bundle targeted to the iOS platform and its respective source map.
```

File: package.json [hunk 11]
```diff
@@ -20,16 +20,20 @@
     "start:ios:prod": "react-native run-ios --configuration Release --scheme 'reactNativeWithMoxy Prod'",
     "start:android:dev": "react-native run-android --variant debug",
     "start:android:prod": "react-native run-android --variant release",
-    "build:android:js:dev": "react-native bundle --platform android --dev true --entry-file index.js --bundle-output ./index.android.bundle",
-    "build:android:js:prod": "react-native bundle --platform android --dev false --entry-file index.js --bundle-output ./index.android.bundle",
-    "build:android:aab:prod": "cd android && ./gradlew bundleRelease || cd -",
+    "bundle:ios:js:prod": "react-native bundle --platform ios --dev false --entry-file index.js --bundle-output index.ios.bundle --sourcemap-output ios.bundle.map",
```
Comment (acostalima) [inline package.json:23]:
```suggestion
    "bundle:ios:js:prod": "react-native bundle --platform ios --dev false --entry-file index.js --bundle-output ./ios/index.ios.bundle --sourcemap-output ios.bundle.map",
```

File: docusaurus/docs/what-is-included/available-scripts.md [hunk 12]
```diff
@@ -24,37 +24,37 @@ This script will launch the app on the first available simulator it is found. Yo
...
 
-You can find the bundle file, `index.android.bundle`, in the project's root directory.
+You can find the source map file, `ios.bundle.map`, in the project's root directory.
```
Comment (acostalima) [inline docusaurus/docs/what-is-included/available-scripts.md:39]:
```suggestion
You will find the the bundle file, `index.ios.bundle`, in the directory `ios/`. The source map file, `ios.bundle.map`, is created at the project's root directory.
```

File: docusaurus/docs/what-is-included/available-scripts.md [hunk 13]
```diff
@@ -24,37 +24,37 @@ This script will launch the app on the first available simulator it is found. Yo
...
 
-## `npm run build:android:js:prod`
+## `npm run bundle:android:js:prod`
 
-This script will create an optimized, production-ready JavaScript bundle targeted to the Android platform.
+This script will create an optimized, production-ready JavaScript bundle targeted to the Android platform and it's respective source map.
```
Comment (acostalima) [inline docusaurus/docs/what-is-included/available-scripts.md:43]:
```suggestion
This script will create an optimized, production-ready JavaScript bundle targeted to the Android platform and its respective source map.
```

**Pull request merged**